Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/DocumentPropertyEditor/PrefabComponentAdapter.cpp

```diff
@@ -0,0 +1,81 @@
+#include <AzToolsFramework/Prefab/Overrides/PrefabOverridePublicInterface.h>
+#include <AzToolsFramework/Prefab/PrefabDomUtils.h>
+#include <AzToolsFramework/Prefab/PrefabPublicInterface.h>
+
+namespace AzToolsFramework::Prefab
+{
+    PrefabComponentAdapter::PrefabComponentAdapter()
+        : ComponentAdapter()
+    {
+        m_prefabOverridePublicInterface = AZ::Interface<AzToolsFramework::Prefab::PrefabOverridePublicInterface>::Get();
+        AZ_Assert(m_prefabOverridePublicInterface, "Could not get PrefabOverridePublicInterface on PrefabComponentAdapter construction.");
+
+        m_prefabPublicInterface = AZ::Interface<AzToolsFramework::Prefab::PrefabPublicInterface>::Get();
+        AZ_Assert(m_prefabPublicInterface, "Could not get PrefabPublicInterface on PrefabComponentAdapter construction.");
+    }
+
+    PrefabComponentAdapter::~PrefabComponentAdapter()
+    {
+    }
+
+    void PrefabComponentAdapter::CreateLabel(
+        AZ::DocumentPropertyEditor::AdapterBuilder* adapterBuilder, AZStd::string_view labelText, AZStd::string_view serializedPath)
+    {
+        using PrefabPropertyEditorNodes::PrefabOverrideLabel;
+
+        adapterBuilder->BeginPropertyEditor<PrefabOverrideLabel>();
+        adapterBuilder->Attribute(PrefabOverrideLabel::Text, labelText);
+
+        AZ::Dom::Path relativePathFromEntity;
+        if (!serializedPath.empty())
+        {
+            relativePathFromEntity /= PrefabDomUtils::ComponentsName;
+            relativePathFromEntity /= m_componentAlias;
+            relativePathFromEntity /= AZ::Dom::Path(serializedPath);
+        }
+
+        // Do not show override visualization on container entities or for empty serialized paths.
+        if (m_prefabPublicInterface->IsInstanceContainerEntity(m_entityId) || relativePathFromEntity.IsEmpty())
+        {
+            adapterBuilder->Attribute(PrefabOverrideLabel::IsOverridden, false);
+        }
+        else
+        {
+            bool isOverridden = m_prefabOverridePublicInterface->AreOverridesPresent(m_entityId, relativePathFromEntity.ToString());
+            adapterBuilder->Attribute(PrefabOverrideLabel::IsOverridden, isOverridden);
+
+            if (isOverridden)
+            {
+                adapterBuilder->Attribute(PrefabOverrideLabel::RelativePath, relativePathFromEntity.ToString());
+                adapterBuilder->AddMessageHandler(this, PrefabOverrideLabel::RevertOverride);
+            }
+        }
+
+        adapterBuilder->EndPropertyEditor();
+    }
+
+    AZ::Dom::Value PrefabComponentAdapter::HandleMessage(const AZ::DocumentPropertyEditor::AdapterMessage& message)
+    {
+        auto handleRevertOverride = [this](AZStd::string_view relativePathFromEntity)
+        {
+            m_prefabOverridePublicInterface->RevertOverrides(m_entityId, relativePathFromEntity);
+        };
+
+        message.Match(PrefabPropertyEditorNodes::PrefabOverrideLabel::RevertOverride, handleRevertOverride);
```

Comment:
Should the return value for this be checked, or does it not return anything?

Reply:
The type is `AZ::Dom::Value`. It currently always returns null since the callback does not return anything.

We can take advantage of it if we want to do something more based on the result.

Reply:
OK. I guess there's no way to know whether the base class ComponentAdapter::HandleMessage still needs to be called? I'm still not super familiar with the DPE message handling.

Reply:
I guess maybe return value of Match() can tell whether the message is matched or not. I am still looking into why it always return null. Maybe @amzn-tmryan knows. But I think it's okay to not check this since base class HandleMessage()s do Match() as well.

Even if we change to handle property edits in PrefabComponentAdapter we still need to call ComponentAdapter::HandleMessage() to "handle" the message. [ReflectionAdapter::HandleMessage()](https://github.com/o3de/o3de/blob/development/Code/Framework/AzToolsFramework/AzToolsFramework/UI/DocumentPropertyEditor/DPEComponentAdapter.cpp#L142) might accept the message.

Reply:
The reason you always see null returned by Match is because the handlers being executed don't return anything:
https://github.com/o3de/o3de/blob/0a666643fd39795fbb33383764017a8ec8e1a4d7/Code/Framework/AzFramework/AzFramework/DocumentPropertyEditor/DocumentSchema.h#L579-L590

If a handler returns a value, then it will be returned by Match.